module/VuFind/src/VuFind/Service/MarkdownFactory.php

```diff
@@ -93,104 +111,194 @@ public function __invoke(
         $requestedName,
         array $options = null
     ) {
-        $markdownConfig = $container->get(\VuFind\Config\PluginManager::class)
-            ->get('markdown');
-
-        $environment = $this->getEnvironment($markdownConfig);
-        $this->addExtensions($environment, $markdownConfig);
+        $this->config = $container->get(\VuFind\Config\PluginManager::class)
+            ->get('markdown')->toArray();
+        $this->extensions = isset($this->config['Markdown']['extensions'])
+            ? array_map(
+                'trim',
+                explode(',', $this->config['Markdown']['extensions'])
+            )
+            : self::$defaultExtensions;
+        $this->extensions = array_filter(
+            $this->extensions,
+            function ($item) {
+                return $item !== '';
+            }
+        );
+        unset($this->config['Markdown']['extensions']);
```

Comment:
Is there a strong reason to unset this here rather than to filter it out when building the configuration below? I confess I haven't studied the code in close detail yet, but just wondering if the same effect could be achieved in a way that relies less on mutable data.

Reply:
This `unset` is even unnecessary. The config is being build from scratch in `getBaseConfig` method, so extensions key is not transferred into final configuration. I've removed it.